Sources/SKCore/BuildSystem.swift

```diff
@@ -147,6 +147,15 @@ public protocol BuildSystem: AnyObject, Sendable {
   /// `nil` if the build system doesn't support topological sorting of targets.
   func topologicalSort(of targets: [ConfiguredTarget]) async -> [ConfiguredTarget]?
 
+  /// Returns the list of targets that might depend on the given target and that need to be re-prepared when a file in
+  /// `target` is modified.
+  ///
+  /// The returned list can be an over-approximation, in which case the indexer will perform more work than strictly
+  /// necessary by scheduling re-preparation of a target where it isn't necessary.
+  ///
+  /// Returning `nil` indicates that all targets should be considered depending on the given target.
+  func targets(dependingOn targets: [ConfiguredTarget]) async -> [ConfiguredTarget]?
```

Comment:
Would taking `Target` make more sense here? We don't really care about the possibly large set of configured targets as input. I'd argue that's even the case for output, but we do need the configured target eventually since that's what we track preparation of.

Reply:
Ah, Alex brought up a very valid point here. It's entirely possible that while the concept of a "target" may contain the file, the underlying target + platform may not - consider settings for including/excluding files per platform. So ConfiguredTarget really is the right abstraction here, thanks for remembering that one :).

Reply:
Added the following comment to `SemanticIndexManager.filesDidChange`

```swift
    // Note that configured targets are the right abstraction layer here (instead of a non-configured target) because a
    // build system might have targets that include different source files. Hence a source file might be in target T
    // configured for macOS but not in target T configured for iOS.
```